src/MainWindow/main_window.cpp

```diff
@@ -181,21 +189,25 @@ void MainWindow::ReShowWindow(QString strProject) {
   runDockWidget->setObjectName("sourcedockwidget");
   RunsForm* runForm = new RunsForm(strProject, this);
   runDockWidget->setWidget(runForm);
-  addDockWidget(Qt::BottomDockWidgetArea, runDockWidget);
 
-  QDockWidget* editorDockWidget = new QDockWidget(this);
-  editorDockWidget->setObjectName("editordockwidget");
   TextEditor* textEditor = new TextEditor(this);
   textEditor->RegisterCommands(GlobalSession);
   textEditor->setObjectName("textEditor");
-  editorDockWidget->setWidget(textEditor->GetTextEditor());
-  addDockWidget(Qt::RightDockWidgetArea, editorDockWidget);
 
   connect(sourForm, SIGNAL(OpenFile(QString)), textEditor,
           SLOT(SlotOpenFile(QString)));
   connect(textEditor, SIGNAL(CurrentFileChanged(QString)), sourForm,
           SLOT(SetCurrentFileItem(QString)));
 
+  QWidget* centralWidget = new QWidget(this);
+  QBoxLayout* layout = new QBoxLayout(QBoxLayout::TopToBottom, centralWidget);
+  layout->setContentsMargins(0, 0, 0, 0);
+  layout->setSpacing(0);
+  layout->addWidget(textEditor->GetTextEditor());
+  centralWidget->setLayout(layout);
+  takeCentralWidget();
```

Comment:
it is doing nothing

Reply:
the next line is
setCentralWidget(centralWidget);

Reply:
so why do you need to call takeCentralWidget(); ?